src/test/java/net/bhardy/bizzo/billing/PolicyBuilderTest.java

```diff
@@ -35,6 +33,84 @@
  */
 public class PolicyBuilderTest {
     @Test
+    public void cycleType_monthlySimple() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .sameDayEveryMonth(5)
+                .build();
+        assertEquals(CycleType.MONTHLY, policy.getCycleType());
+    }
+
+    @Test
+    public void cycleType_monthlyFiltered() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .sameDayEveryMonth(5)
+                .filter(PolicyFilter.not(PolicyFilter.daysOfWeek(WEDNESDAY)))
+                .action(ActionChoice.Kind.NEXT_DAY)
+                .build();
+        assertEquals(CycleType.MONTHLY, policy.getCycleType());
+    }
+
+    @Test
+    public void cycleType_weeklySimple() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .sameDayEveryWeek(FRIDAY)
+                .build();
+        assertEquals(CycleType.WEEKLY, policy.getCycleType());
+    }
+
+    @Test
+    public void cycleType_weeklyFiltered() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .sameDayEveryWeek(FRIDAY)
+                .filter(PolicyFilter.not(PolicyFilter.daysOfWeek(WEDNESDAY)))
+                .action(ActionChoice.Kind.NEXT_DAY)
+                .build();
+        assertEquals(CycleType.WEEKLY, policy.getCycleType());
+    }
+
+    @Test
+    public void cycleType_dailySimple() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .everyDay()
+                .build();
+        assertEquals(CycleType.DAILY, policy.getCycleType());
+    }
+
+    @Test
+    public void cycleType_dailyFiltered() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .everyDay()
+                .filter(PolicyFilter.not(PolicyFilter.daysOfWeek(WEDNESDAY)))
+                .action(ActionChoice.Kind.NEXT_DAY)
+                .build();
+        assertEquals(CycleType.DAILY, policy.getCycleType());
+    }
+
+    @Test
+    public void isDueOn_weeklySimple() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .sameDayEveryWeek(SATURDAY)
+                .build();
+
+        LocalDate oneSaturday = LocalDate.of(2017, 8, 5);
+        assertTrue(policy.isDueOn(oneSaturday));
+    }
+
+    @Test
+    public void isDueOn_weeklyFiltered() {
+        BillingPolicy policy = BillingPolicy.builder()
+                .sameDayEveryWeek(FRIDAY)
+                .filter(PolicyFilter.not(PolicyFilter.daysOfWeek(SATURDAY, SUNDAY)))
+                .action(ActionChoice.Kind.NEXT_DAY)
+                .build();
+
+        LocalDate oneSaturday = LocalDate.of(2017, 8, 5);
+        assertFalse(policy.isDueOn(oneSaturday));
+        LocalDate monday = LocalDate.of(2017, 8, 7);
+        assertFalse(policy.isDueOn(monday));
+    }
+
+    @Test
     public void isDueOn_monthlySimple() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)